ledger-core/v2/virtual/handlers/vdelegatedcallrequest.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2020 Insolar Network Ltd.
+// All rights reserved.
+// This material is licensed under the Insolar License version 1.0,
+// available at https://github.com/insolar/assured-ledger/blob/master/LICENSE.md.
+
+package handlers
+
+import (
+	"context"
+
+	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor"
+	"github.com/insolar/assured-ledger/ledger-core/v2/conveyor/smachine"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/contract"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/jet"
+	"github.com/insolar/assured-ledger/ledger-core/v2/insolar/payload"
+	"github.com/insolar/assured-ledger/ledger-core/v2/log"
+	"github.com/insolar/assured-ledger/ledger-core/v2/network/messagesender"
+	messageSenderAdapter "github.com/insolar/assured-ledger/ledger-core/v2/network/messagesender/adapter"
+	"github.com/insolar/assured-ledger/ledger-core/v2/pulse"
+	"github.com/insolar/assured-ledger/ledger-core/v2/reference"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/injector"
+	"github.com/insolar/assured-ledger/ledger-core/v2/vanilla/throw"
+	"github.com/insolar/assured-ledger/ledger-core/v2/virtual/object"
+)
+
+var deadBeef = [...]byte{0xde, 0xad, 0xbe, 0xef}
+
+type SMVDelegatedCallRequest struct {
+	// input arguments
+	Meta    *payload.Meta
+	Payload *payload.VDelegatedCallRequest
+
+	node reference.Global
+
+	objectSharedState object.SharedStateAccessor
+
+	// dependencies
+	objectCatalog     object.Catalog
+	messageSender     messageSenderAdapter.MessageSender
+	jetAffinityHelper jet.AffinityHelper
```

Comment:
jet.AffinityHelper should not be directly present in SM
please isolate via an adapter/service

Reply:
We've discussed this with @bigbes, it should be onde in separate PR. New adapter/service should be introduced and this helper should be removed.